src/main/diagnostics/diagnostics.ts

```diff
@@ -1,68 +1,121 @@
-import DIAGNOSTICS_TABLE from '@/assets/diagnostics.yaml';
+export enum DiagnosticsCategory {
+  Utilities = 'Utilities',
+  Networking = 'Networking',
+}
+
+/**
+ * DiagnosticsChecker describes an implementation of a single diagnostics check.
+ */
+export interface DiagnosticsChecker {
+  /** Unique identifier for this check. */
+  id: string;
+  documentation: string,
+  description: string,
+  category: DiagnosticsCategory,
+  /**
+   * Preform the check.  Returns true if the check passed (i.e. things are
+   * working correctly and does not need to be fixed).
+   */
+  check(): Promise<boolean>;
+}
 
 type DiagnosticsFix = {
-  fixes: { description: string }
+  description: string
 };
 
-export type DiagnosticsCheck = {
+export type DiagnosticsResult = {
   id: string,
```

Comment:
I'll go for option 1.
Option 2 is unfeasible, because this structure needs to be JSON-serializable (to get passed to the front end).  The checker includes a method, so it can't be serialized.

Reply:
I just realized that we _can't_ rename (I had it as `checkerId`), because this structure is what is emitter over JSON (see `command-api.yaml`).  Or rather, we could but then it doesn't make any sense from an API point of view (since there's no concept of checkers as far as the API is concerned).

Reply:
Ok that's fair. As long as we have comments in place saying what `id` is for (which I see we do) we should be good.